api/src/main/java/org/pmiops/workbench/api/DataBrowserController.java

```diff
@@ -0,0 +1,292 @@
+        }
+        else if ((stratum1 != null && !stratum1.isEmpty())) {
+            resultList = analysisResultDao.findAnalysisResultsByAnalysisIdAndStratum1(analysisId, stratum1);
+        }
+        else {
+            resultList = analysisResultDao.findAnalysisResultsByAnalysisId(analysisId);
+        }
+
+        AnalysisResultListResponse resp = new AnalysisResultListResponse();
+        resp.setItems(resultList.stream().map(TO_CLIENT_ANALYSIS_RESULT).collect(Collectors.toList()));
+
+
+        return ResponseEntity.ok(resp);
+    }
+
+    @Override
+    public ResponseEntity<org.pmiops.workbench.model.AnalysisResult> getParticipantCount() {
+        long aid =  1; // partipant count analysis id
+        final List<AnalysisResult> resultList  =  analysisResultDao.findAnalysisResultsByAnalysisId(aid);
+        AnalysisResultListResponse resp = new AnalysisResultListResponse();
+        resp.setItems(resultList.stream().map(TO_CLIENT_ANALYSIS_RESULT).collect(Collectors.toList()));
+        if ( resp.getItems().size() > 0 ) {
+            return ResponseEntity.ok(resp.getItems().get(0));
+        }
+        else {
+            // Todo handle error better
+            org.pmiops.workbench.model.AnalysisResult r = new org.pmiops.workbench.model.AnalysisResult();
+            r.setAnalysisId( (long) 1);
+            r.setCountValue((long) 0);
+            return ResponseEntity.ok(r);
+        }
+    }
+
+    @Override
+    public ResponseEntity<AnalysisResultListResponse> getConceptCount(String conceptId) {
+        long aid = 3000;
+        return this.getAnalysisResults(aid, conceptId, null, null , null, null);
+    }
+
+    @Override
+    public ResponseEntity<AnalysisResultListResponse> getConceptCountByGender(String conceptId) {
+        long aid = 3101;
+        return this.getAnalysisResults(aid, conceptId, null, null,null,null);
+    }
+
+    @Override
+    public ResponseEntity<AnalysisResultListResponse> getConceptCountByAge(String conceptId) {
+        long aid = 3102;
+        return this.getAnalysisResults(aid, conceptId, null, null , null, null);
+    }
+
+    @Override
+    public ResponseEntity<AnalysisListResponse> getAnalyses() {
+        long num = 1;
+        final List<AchillesAnalysis> resultList = achillesAnalysisDao.findAllByAnalysisIdIsGreaterThanEqual((Long)num);
+        AnalysisListResponse resp = new AnalysisListResponse();
+        resp.setItems(resultList.stream().map(TO_CLIENT_ANALYSIS).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+
+    }
+
+    @Override
+    public ResponseEntity<DbDomainListResponse> getDbDomains() {
+        final List<DbDomain> resultList = dbDomainDao.findAllByDomainIdIsNotNull();
+        DbDomainListResponse resp = new DbDomainListResponse();
+        resp.setItems(resultList.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
```

Comment:
Can we write some mock unit test cases for this Controller?

Reply:
We should not be using lots of data to do unit tests. We should just have tests that insert data into the tables (in memory) and verify the controllers do the right thing, like we do in other places, e.g.:

https://github.com/all-of-us/workbench/blob/master/api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java